### ISSUE TITLE
namedMock does not use global namespace in function parameter type hints

### ISSUE THREAD

**Issue by nigelchanyk** [mockery/mockery#396]
``` php
<?php

namespace Foo {

  class Bar {
    public function value() {
      return 0;
    }
  }

}

namespace {

  class Meh {
    public function doSomething(\Foo\Bar $bar) {
      return $bar->value();
    }
  }

  $meh = \Mockery::namedMock("ILoveRandomNamespace\\Meh", "Meh");
  $bar = new \Foo\Bar();
  $meh->shouldReceive('doSomething')->once()->andReturn(1337);
  var_dump($meh->doSomething($bar));
}
```

Error:
Catchable fatal error: Argument 1 passed to ILoveRandomNamespace\Meh::doSomething() must be an instance of ILoveRandomNamespace\Foo\Bar, instance of Foo\Bar given, called in /home/nchan/meh.php on line 24 and defined in /data/users/fbcode/hphp/test/frameworks/framework_downloads/mockery/library/Mockery/Loader/EvalLoader.php(16) : eval()'d code on line 704

Cause:
The eval()'d code has namespace ILoveRandomNamespace. The generated doSomething() function in the eval()'d code accepts parameter Foo\Bar. Since we're still in ILoveRandomNamespace, ILoveRandomNamespace\Foo\Bar is inferred.

**Comment by robertbasic** [comment]
Assigning to myself as I tried a PR for this a long time ago...

**Comment by robertbasic** [comment]
Same as #525, this issue was fixed with PR #558 

Closing as fixed.
